tests/string.cpp

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2012-present Facebook, Inc.
+ * Licensed under the Apache License, Version 2.0 */
+
+#include "watchman.h"
+#include <algorithm>
+#include "thirdparty/tap.h"
+
+void w_log(int level, WATCHMAN_FMT_STRING(const char *fmt), ...) {
+  unused_parameter(level);
+  unused_parameter(fmt);
+}
+
+void test_suffix() {
+
+  w_string_t *suffix = nullptr, *str = nullptr, *expect_suffix = nullptr;
+  str = w_string_new_typed("", W_STRING_UNICODE);
```

Comment:
prefer to use the newer smart pointer type for this:

```
w_string str("");
```

Note that it doesn't make too much sense to use `W_STRING_UNICODE` in here.  That is a tag to help the serialization layer recreate string or binary representations as appropriate.

Filenames in watchman are always binary which happens to be the default type when constructing a string via the `w_string` constructor, so you can just omit it.